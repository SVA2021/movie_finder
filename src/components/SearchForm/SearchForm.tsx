import CloseIcon from '@mui/icons-material/Close';
import {Box, Button, Grid, MenuItem, Select, Slider, Typography} from '@mui/material';
import {FC, memo, useEffect} from 'react';
import {Controller, SubmitHandler, useForm} from 'react-hook-form';
import {useAppDispatch, useAppSelector} from '../../app/hooks';
import {selectFilters, setSearchExtended} from '../../features/movieFinder/movieFinderSlice';
import {getFiltersAsync} from '../../features/movieFinder/movieFinderThunks';
import {TSearchForm} from '../../features/movieFinder/movieFinderTypes';
interface SearchFormProps {
  initialFilters: TSearchForm | null
  closeHandler: () => void
}

const DEFAULT_SEARCH_FORM: TSearchForm = {
  countries: 'default',
  genres: 'default',
  order: 'RATING',
  rating: [0, 10],
  year: [1950, 2024],
}

export const SearchForm: FC<SearchFormProps> = memo(({closeHandler, initialFilters}) => {

  const dispatch = useAppDispatch();
  const Filters = useAppSelector(selectFilters);
  const Countries = Filters?.countries ?? [];
  const Genres = Filters?.genres ?? [];
  const BASE_FILTER_DATA: TSearchForm = initialFilters ?? DEFAULT_SEARCH_FORM;

  const onSubmit: SubmitHandler<TSearchForm> = data => onSubmitHandler(data);
  const {control, handleSubmit} = useForm<TSearchForm>();

  function onSubmitHandler(data: TSearchForm) {
    dispatch(setSearchExtended(data));
    closeHandler();
  }

  useEffect(() => {
    if (Filters === null) dispatch(getFiltersAsync());
  }, []);

  return (
    <Box
      p={{xs: 1, md: 2, lg: 4}}
      zIndex={1000}
      bgcolor={'primary.main'}
      sx={{
        position: 'fixed',
        inset: 0,
        height: '100vh',
        width: '100vw',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
      }}
    >
      <Box component={'form'} onSubmit={handleSubmit(onSubmit)} width={'80%'} >
        <Typography mb={{xs: 4, md: 20}} variant="h3" color={'secondary'} textAlign={'center'} >
          Дополнительные фильтры поиска
        </Typography>
        <Grid container spacing={{xs: 0, md: 8}} >
          <Grid item xs={12} md={6}>
            <Typography pb={4} variant="body1" component={'h4'} textTransform={'capitalize'} color={'secondary'}>
              год выпуска
            </Typography>
            <Controller
              name={'year'}
              control={control}
              defaultValue={BASE_FILTER_DATA.year}
              render={({field}) => (
                <Slider
                  {...field}
                  onChange={(_, value) => field.onChange(value)}
                  valueLabelDisplay="on"
                  max={2022}
                  min={1950}
                  step={1}
                  marks
                  color={'secondary'}
                />
              )}
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <Typography pb={4} variant="body1" component={'h4'} textTransform={'capitalize'} color={'secondary'}>
              рейтинг
            </Typography>
            <Controller
              name={'rating'}
              control={control}
              defaultValue={BASE_FILTER_DATA.rating}
              render={({field}) => (
                <Slider
                  {...field}
                  onChange={(_, value) => field.onChange(value)}
                  valueLabelDisplay="on"
                  max={10}
                  min={0}
                  step={0.5}
                  marks
                  color={'secondary'}
                />
              )}
            />
          </Grid>

          <Grid item xs={12} md={4}>
            <Typography pb={4} variant="body1" component={'h4'} color={'secondary'}>
              Сортировать по
            </Typography>
            <Controller
              render={({field}) => (
                <Select {...field} color={'secondary'} >
                  <MenuItem value={'RATING'}>рейтингу</MenuItem>
                  <MenuItem value={'NUM_VOTE'}>количеству голосов</MenuItem>
                  <MenuItem value={'YEAR'}>году выпуска</MenuItem>
                </Select>
              )}
              defaultValue={BASE_FILTER_DATA.order}
              name={'order'}
              control={control}
            />
          </Grid>

          <Grid item xs={12} md={4}>
            <Typography pb={4} variant="body1" component={'h4'} color={'secondary'}>
              Страна производитель
            </Typography>
            <Controller
              render={({field}) => (
                <Select {...field} color={'secondary'} >
                  <MenuItem value={'default'}>сбросить фильтр</MenuItem>
                  {
                    Countries.map((item) =>
                      <MenuItem key={item.id} value={item.id}>{item.country}</MenuItem>
                    )
                  }
                </Select>
              )}
              defaultValue={BASE_FILTER_DATA.countries}
              name={'countries'}
              control={control}
            />
          </Grid>

          <Grid item xs={12} md={4}>
            <Typography pb={4} variant="body1" component={'h4'} color={'secondary'}>
              Жанр
            </Typography>
            <Controller
              render={({field}) => (
                <Select {...field} color={'secondary'} >
                  <MenuItem value={'default'}>сбросить фильтр</MenuItem>
                  {
                    Genres.map((item) =>
                      <MenuItem key={item.id} value={item.id}>{item.genre}</MenuItem>
                    )
                  }
                </Select>
              )}
              defaultValue={BASE_FILTER_DATA.genres}
              name={'genres'}
              control={control}
            />
          </Grid>
        </Grid>

        <Box mt={4} textAlign={'center'} >
          <Button type={'submit'} >Искать</Button>
        </Box>

      </Box>

      <Box sx={{position: 'absolute', top: {xs: 10, md: 32}, right: {xs: 10, md: 32}}}>
        <Button onClick={closeHandler} ><CloseIcon fontSize='small' /></Button>
      </Box>

    </Box>
  );
});