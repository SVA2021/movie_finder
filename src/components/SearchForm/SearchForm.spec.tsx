import {fireEvent, render, screen} from "@testing-library/react";
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import {store} from "../../app/store";
import {SearchForm} from "./SearchForm";

describe('render SearchForm', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <SearchForm initialFilters={null} />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('search-form')).toMatchSnapshot();
  });

})