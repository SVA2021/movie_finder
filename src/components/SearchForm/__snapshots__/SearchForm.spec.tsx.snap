// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SearchForm snapshot test 1`] = `
<form
  class="MuiBox-root css-1x97c6v"
  data-testid="search-form"
>
  <h3
    class="MuiTypography-root MuiTypography-h3 css-1emirj-MuiTypography-root"
  >
    Дополнительные фильтры поиска
  </h3>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-md-8 css-enkqwv-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-body1 css-16n20ao-MuiTypography-root"
      >
        год выпуска
      </h4>
      <span
        class="MuiSlider-colorSecondary MuiSlider-sizeMedium MuiSlider-root css-1diafny-MuiSlider-root"
      >
        <span
          class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
        />
        <span
          class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
          style="left: 0%; width: 100%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="0"
          style="left: 0%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="1"
          style="left: 1.3888888888888888%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="2"
          style="left: 2.7777777777777777%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="3"
          style="left: 4.166666666666667%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="4"
          style="left: 5.555555555555555%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="5"
          style="left: 6.944444444444445%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="6"
          style="left: 8.333333333333334%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="7"
          style="left: 9.722222222222221%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="8"
          style="left: 11.11111111111111%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="9"
          style="left: 12.5%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="10"
          style="left: 13.88888888888889%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="11"
          style="left: 15.277777777777779%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="12"
          style="left: 16.666666666666668%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="13"
          style="left: 18.055555555555557%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="14"
          style="left: 19.444444444444443%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="15"
          style="left: 20.833333333333332%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="16"
          style="left: 22.22222222222222%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="17"
          style="left: 23.61111111111111%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="18"
          style="left: 25%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="19"
          style="left: 26.38888888888889%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="20"
          style="left: 27.77777777777778%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="21"
          style="left: 29.166666666666668%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="22"
          style="left: 30.555555555555557%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="23"
          style="left: 31.944444444444443%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="24"
          style="left: 33.333333333333336%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="25"
          style="left: 34.72222222222222%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="26"
          style="left: 36.111111111111114%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="27"
          style="left: 37.5%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="28"
          style="left: 38.888888888888886%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="29"
          style="left: 40.27777777777778%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="30"
          style="left: 41.666666666666664%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="31"
          style="left: 43.05555555555556%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="32"
          style="left: 44.44444444444444%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="33"
          style="left: 45.833333333333336%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="34"
          style="left: 47.22222222222222%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="35"
          style="left: 48.611111111111114%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="36"
          style="left: 50%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="37"
          style="left: 51.388888888888886%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="38"
          style="left: 52.77777777777778%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="39"
          style="left: 54.166666666666664%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="40"
          style="left: 55.55555555555556%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="41"
          style="left: 56.94444444444444%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="42"
          style="left: 58.333333333333336%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="43"
          style="left: 59.72222222222222%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="44"
          style="left: 61.111111111111114%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="45"
          style="left: 62.5%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="46"
          style="left: 63.888888888888886%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="47"
          style="left: 65.27777777777777%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="48"
          style="left: 66.66666666666667%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="49"
          style="left: 68.05555555555556%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="50"
          style="left: 69.44444444444444%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="51"
          style="left: 70.83333333333333%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="52"
          style="left: 72.22222222222223%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="53"
          style="left: 73.61111111111111%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="54"
          style="left: 75%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="55"
          style="left: 76.38888888888889%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="56"
          style="left: 77.77777777777777%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="57"
          style="left: 79.16666666666667%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="58"
          style="left: 80.55555555555556%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="59"
          style="left: 81.94444444444444%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="60"
          style="left: 83.33333333333333%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="61"
          style="left: 84.72222222222223%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="62"
          style="left: 86.11111111111111%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="63"
          style="left: 87.5%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="64"
          style="left: 88.88888888888889%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="65"
          style="left: 90.27777777777777%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="66"
          style="left: 91.66666666666667%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="67"
          style="left: 93.05555555555556%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="68"
          style="left: 94.44444444444444%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="69"
          style="left: 95.83333333333333%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="70"
          style="left: 97.22222222222223%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="71"
          style="left: 98.61111111111111%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="72"
          style="left: 100%;"
        />
        <span
          class="MuiSlider-thumbColorSecondary MuiSlider-thumbSizeMedium MuiSlider-thumb css-frv3qg-MuiSlider-thumb"
          data-index="0"
          style="left: 0%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="2022"
            aria-valuemin="1950"
            aria-valuenow="1950"
            data-index="0"
            max="2022"
            min="1950"
            name="year"
            step="1"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="1950"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-17djbj3-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                1950
              </span>
            </span>
          </span>
        </span>
        <span
          class="MuiSlider-thumbColorSecondary MuiSlider-thumbSizeMedium MuiSlider-thumb css-frv3qg-MuiSlider-thumb"
          data-index="1"
          style="left: 100%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="2022"
            aria-valuemin="1950"
            aria-valuenow="2022"
            data-index="1"
            max="2022"
            min="1950"
            name="year"
            step="1"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="2022"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-17djbj3-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                2022
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-body1 css-16n20ao-MuiTypography-root"
      >
        рейтинг
      </h4>
      <span
        class="MuiSlider-colorSecondary MuiSlider-sizeMedium MuiSlider-root css-1diafny-MuiSlider-root"
      >
        <span
          class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
        />
        <span
          class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
          style="left: 0%; width: 100%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="0"
          style="left: 0%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="1"
          style="left: 5%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="2"
          style="left: 10%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="3"
          style="left: 15%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="4"
          style="left: 20%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="5"
          style="left: 25%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="6"
          style="left: 30%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="7"
          style="left: 35%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="8"
          style="left: 40%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="9"
          style="left: 45%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="10"
          style="left: 50%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="11"
          style="left: 55%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="12"
          style="left: 60%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="13"
          style="left: 65%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="14"
          style="left: 70%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="15"
          style="left: 75%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="16"
          style="left: 80%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="17"
          style="left: 85%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="18"
          style="left: 90%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="19"
          style="left: 95%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="20"
          style="left: 100%;"
        />
        <span
          class="MuiSlider-thumbColorSecondary MuiSlider-thumbSizeMedium MuiSlider-thumb css-frv3qg-MuiSlider-thumb"
          data-index="0"
          style="left: 0%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="10"
            aria-valuemin="0"
            aria-valuenow="0"
            data-index="0"
            max="10"
            min="0"
            name="rating"
            step="0.5"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="0"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-17djbj3-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                0
              </span>
            </span>
          </span>
        </span>
        <span
          class="MuiSlider-thumbColorSecondary MuiSlider-thumbSizeMedium MuiSlider-thumb css-frv3qg-MuiSlider-thumb"
          data-index="1"
          style="left: 100%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="10"
            aria-valuemin="0"
            aria-valuenow="10"
            data-index="1"
            max="10"
            min="0"
            name="rating"
            step="0.5"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="10"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-17djbj3-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                10
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-body1 css-z56ql5-MuiTypography-root"
      >
        Сортировать по
      </h4>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorSecondary  css-es1mxv-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="mui-component-select-order"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="mui-component-select-order"
          role="button"
          tabindex="0"
        >
          рейтингу
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          name="order"
          tabindex="-1"
          value="RATING"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-hdw1oc"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-body1 css-z56ql5-MuiTypography-root"
      >
        Страна производитель
      </h4>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorSecondary  css-es1mxv-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="mui-component-select-countries"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="mui-component-select-countries"
          role="button"
          tabindex="0"
        >
          сбросить фильтр
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          name="countries"
          tabindex="-1"
          value="default"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-hdw1oc"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-body1 css-z56ql5-MuiTypography-root"
      >
        Жанр
      </h4>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorSecondary  css-es1mxv-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="mui-component-select-genres"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="mui-component-select-genres"
          role="button"
          tabindex="0"
        >
          сбросить фильтр
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          name="genres"
          tabindex="-1"
          value="default"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-hdw1oc"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-4sywta"
  >
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Искать
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</form>
`;
